src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -944,6 +945,116 @@ public function providePathsForIsAbsolutePath()
         );
     }
 
+    public function testTempnam()
+    {
+        $dirname = $this->workspace;
+
+        $filename = $this->filesystem->tempnam($dirname, 'foo');
+
+        $this->assertNotFalse($filename);
+        $this->assertFileExists($filename);
+    }
+
+    public function testTempnamWithFileScheme()
+    {
+        $scheme = 'file://';
+        $dirname = $scheme.$this->workspace;
+
+        $filename = $this->filesystem->tempnam($dirname, 'foo');
+
+        $this->assertNotFalse($filename);
+        $this->assertStringStartsWith($scheme, $filename);
+        $this->assertFileExists($filename);
+    }
+
+    public function testTempnamWithMockScheme()
+    {
+        // We avoid autoloading via ClassLoader as stream_wrapper_register creates the object
+        if (!@include __DIR__.DIRECTORY_SEPARATOR.'Fixtures'.DIRECTORY_SEPARATOR.'MockStream'.DIRECTORY_SEPARATOR.'MockStream.php') {
```

Comment:
I think the issue might have been because the file wasn't using PSR-0 namespaces. I have removed the include and now rely on autoloading to load the class